test/env/data.go

```diff
@@ -209,21 +210,50 @@ func CheckBundleSynced(ctx context.Context, cl client.Client, bundleName string,
 	})
 }
 
-// CheckBundleSyncedStartsWith is similar to CheckBundleSynced but only checks that the synced bundle starts with the given data,
+// CheckBundleSyncedContains is similar to CheckBundleSynced but only checks that the synced bundle contains the given data,
 // along with checking that the rest of the data contains at least one valid certificate
-func CheckBundleSyncedStartsWith(ctx context.Context, cl client.Client, name string, namespace string, startingData string) error {
+func CheckBundleSyncedContains(ctx context.Context, cl client.Client, name string, namespace string, containedData string) error {
 	return checkBundleSyncedInternal(ctx, cl, name, namespace, func(got string) error {
-		if !strings.HasPrefix(got, startingData) {
-			return fmt.Errorf("received data didn't start with expected data")
+		var block *pem.Block
+		certBytes := []byte(containedData)
+
+	LOOP:
+		for {
+			block, certBytes = pem.Decode(certBytes)
+			if block == nil {
+				break LOOP
+			}
+
+			if block.Type != "CERTIFICATE" {
+				return fmt.Errorf("couldn't decode PEM block containing certificate")
+			}
+
+			if !(strings.Contains(got, string(bytes.Trim(pem.EncodeToMemory(block), "\n")))) {
+				return fmt.Errorf("did not find all certs")
+			}
 		}
 
-		remaining := strings.TrimPrefix(got, startingData)
-
+		certBytes = []byte(got)
 		// check that there are a nonzero number of valid certs remaining
+		found := false
+	LOOP2:
+		for {
+			block, certBytes = pem.Decode(certBytes)
+			if block == nil {
+				break LOOP2
```

Comment:
Can't this be replaced with `continue`? It seems like the purpose of this `break` is to jump to the next iteration of the loop. I don't see why `break LOOP2` is needed.

Reply:
Probably. I just copied the almost identical deduplication code.

Reply:
I just checked. This needs to be break. This will consume the buffer and the loop has to stop once it ran out of bytes.

Reply:
It probably is. I will change that.

Reply:
Changed it in my (copied) version and in the original place where I got it from.